### PR TITLE
[BOLT] Extend calculateEmittedSize() for block size calculation

### PR DESCRIPTION
**Pull request by ShatianWang**
This commit modifies BinaryContext::calculateEmittedSize to update the BinaryBasicBlock::OutputAddressRange for each basic block in the input BF. The modification is done in place, where BinaryBasicBlock::getOutputSize() now gives the emitted size of the basic block.

### CODE REVIEW

**Comment by ShatianWang** [comment]
Note: this PR is 1/8 for upstreaming a new 3-way (hot-warm-cold) function splitting algorithm CDSplit.

File: bolt/include/bolt/Core/BinaryContext.h [hunk 1]
```diff
@@ -1230,6 +1230,9 @@ class BinaryContext {
   ///
   /// Return the pair where the first size is for the main part, and the second
   /// size is for the cold one.
+  /// Modify BinaryBasicBlock::OutputAddressRange for each basic block in the
+  /// function in place so that BB.OutputAddressRange.second less
+  /// BB.OutputAddressRange.first gives the emitted size of BB.
```
Comment (maksfb) [inline bolt/include/bolt/Core/BinaryContext.h:1235]: "... BasicBlock::getOutputSize() gives the emitted size of BB."

Please update the summary as well.

**Comment by ShatianWang** [comment]
Based on comments by @maksfb: 
- [Fix-up 1](https://github.com/llvm/llvm-project/pull/73076/commits/ac9dd53386aaa64db1bc8ec1fe56149753b3598d) modified `BinaryContext::calculateEmittedSize`
- [fixup! Fix-up 2 -- adding a test](https://github.com/llvm/llvm-project/pull/73076/commits/5b10432d0e1a479d906f898747b08ac75b70dbd4) added a test
- [fixup! fixup! Fix-up 2 -- adding a test](https://github.com/llvm/llvm-project/pull/73076/commits/ce86547bb705727c2b4ac3d2cdac008491aef575) modified the printed BB output range format into `Output Address Range: [0x..., 0x....) (N bytes)`
- [fixup! fixup! fixup! Fix-up 2 -- adding a test](https://github.com/llvm/llvm-project/pull/73076/commits/59c14246c1ff61a4a2567a8658813fa8f2e86f15) used `formatv` to implement the new format to make the code cleaner

**Review by maksfb — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**